### PR TITLE
Dont put geneids on session object

### PR DESCRIPTION
**Pull request by dnil**
### This PR adds | fixes:
- (Scout) session issues with large gene lists

**How to prepare for test**:
- [ ] `ssh` to ...
- [ ] Install on stage:
`bash servers/resources/SERVER.scilifelab.se/update-[THIS_TOOL]-stage.sh [THIS-BRANCH-NAME]`

### How to test:
- add genes and see that both report and pdf conversion still work
- add many genes and note that you are no longer kicked out of your session

### Expected outcome:
- [ ]

### Review:
- [ ] Code approved by
- [ ] Tests executed by
- [ ] "Merge and deploy" approved by

This [version](https://semver.org/) is a:
- [ ] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

### CODE REVIEW

**Review by northwestwitch — COMMENTED** [review]
I tried testing the branch, but I think it might be based on some oldish code?

```
(chanjo) chiararasi@ChiaraRMBP:~/Documents/work/GITs/chanjo-report$ make report
docker-compose run -p 5000:5000 chanjo-report chanjo -d mysql+pymysql://chanjoUser:chanjoPassword@mariadb/chanjo4_test report --render html
Creating chanjo-report_chanjo-report_run ... done
Traceback (most recent call last):
  File "/usr/local/bin/chanjo", line 8, in <module>
    sys.exit(root())
  File "/usr/local/lib/python3.8/site-packages/click/core.py", line 1130, in __call__
    return self.main(*args, **kwargs)
  File "/usr/local/lib/python3.8/site-packages/click/core.py", line 1055, in main
    rv = self.invoke(ctx)
  File "/usr/local/lib/python3.8/site-packages/click/core.py", line 1651, in invoke
    cmd_name, cmd, args = self.resolve_command(ctx, args)
  File "/usr/local/lib/python3.8/site-packages/click/core.py", line 1698, in resolve_command
    cmd = self.get_command(ctx, cmd_name)
  File "/usr/local/lib/python3.8/site-packages/chanjo/cli/base.py", line 39, in get_command
    return commands[name].load()
  File "/usr/local/lib/python3.8/site-packages/pkg_resources/__init__.py", line 2450, in load
    return self.resolve()
  File "/usr/local/lib/python3.8/site-packages/pkg_resources/__init__.py", line 2456, in resolve
    module = __import__(self.module_name, fromlist=['__name__'], level=0)
  File "/home/worker/app/chanjo_report/cli/__init__.py", line 2, in <module>
    from .core import report
  File "/home/worker/app/chanjo_report/cli/core.py", line 4, in <module>
    from chanjo_report.interfaces import html, pdf
  File "/home/worker/app/chanjo_report/interfaces/__init__.py", line 2, in <module>
    from .html import render_html
  File "/home/worker/app/chanjo_report/interfaces/html.py", line 4, in <module>
    from chanjo_report.server.app import create_app
  File "/home/worker/app/chanjo_report/server/app.py", line 5, in <module>
    from .config import DefaultConfig
  File "/home/worker/app/chanjo_report/server/config.py", line 4, in <module>
    from .blueprints import index_bp, report_bp
  File "/home/worker/app/chanjo_report/server/blueprints/__init__.py", line 2, in <module>
    from .index import index_bp
  File "/home/worker/app/chanjo_report/server/blueprints/index/__init__.py", line 2, in <module>
    from .views import index_bp
  File "/home/worker/app/chanjo_report/server/blueprints/index/views.py", line 7, in <module>
    from chanjo_report.server.extensions import api
  File "/home/worker/app/chanjo_report/server/extensions.py", line 8, in <module>
    api = Alchy(Model=BASE)
  File "/usr/local/lib/python3.8/site-packages/flask_alchy.py", line 28, in __init__
    super(Alchy, self).__init__(
  File "/usr/local/lib/python3.8/site-packages/flask_sqlalchemy/__init__.py", line 738, in __init__
    session_options.setdefault('scopefunc', connection_stack.__ident_func__)
AttributeError: '_FakeStack' object has no attribute '__ident_func__'
ERROR: 1
make: *** [report] Error 1
```

File: chanjo_report/server/blueprints/report/views.py [hunk 1]
```diff
@@ -124,14 +124,14 @@ def json_gene_coverage():
     return jsonify(results)
 
 
-@report_bp.route("/report", methods=["GET", "POST"])
+@report_bp.route("/report", methods=["POST", "GET"])
```
[the remark this answers is not in the document]
Comment (dnil) [reply chanjo_report/server/blueprints/report/views.py:127]: nono, just a manual revert!

[the remark this answers is not in the document]
Comment (dnil) [reply chanjo_report/server/blueprints/report/views.py:127]: I originally blocked GET completely, but had to agree with you that it changes the API. 😁

File: chanjo_report/server/blueprints/report/controllers.py [hunk 2]
```diff
@@ -7,23 +7,19 @@
...
     Args:
         request(flask.Request)
 
     Returns:
         data(dict)
     """
 
-    sample_ids = request.args.getlist("sample_id") or request.form.getlist("sample_id")
-    raw_gene_ids = request.args.get("gene_ids") or request.form.get("gene_ids")
+    sample_ids = request.form.getlist("sample_id") or request.args.getlist("sample_id")
```
[the remark this answers is not in the document]
Comment (dnil) [reply chanjo_report/server/blueprints/report/controllers.py:18]: No, same same - itty bitty performance diff given that most requests will be POST.

File: chanjo_report/server/blueprints/report/views.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (northwestwitch) [reply chanjo_report/server/blueprints/report/views.py:127]: Ah I see!

**Comment by northwestwitch** [comment]
Tip: to launch a server with demo data:

1. make build
2. make setup
3. make report

**Comment by dnil** [comment]
> Tip: to launch a server with demo data:
> 
> 1. make build
> 2. make setup
> 3. make report

But then I dont make use of my local mysqld! 😉 But good, I'll conform.

**Comment by northwestwitch** [comment]
> But then I dont make use of my local mysqld! 😉 But good, I'll conform.

Whatever works for testing is fine by me :)

**Comment by dnil** [comment]
The Docker version fails for me (both on master and this branch):
```
➜  chanjo-report git:(master) ✗ make report
docker-compose run -p 5000:5000 chanjo-report chanjo -d mysql+pymysql://chanjoUser:chanjoPassword@mariadb/chanjo4_test report --render html
[+] Running 1/0
 ⠿ Container mariadb  Running                                                                                                                                                                       0.0s
Traceback (most recent call last):
  File "/venv/bin/chanjo", line 8, in <module>
    sys.exit(root())
  File "/venv/lib/python3.8/site-packages/click/core.py", line 1130, in __call__
    return self.main(*args, **kwargs)
  File "/venv/lib/python3.8/site-packages/click/core.py", line 1055, in main
    rv = self.invoke(ctx)
  File "/venv/lib/python3.8/site-packages/click/core.py", line 1651, in invoke
    cmd_name, cmd, args = self.resolve_command(ctx, args)
  File "/venv/lib/python3.8/site-packages/click/core.py", line 1698, in resolve_command
    cmd = self.get_command(ctx, cmd_name)
  File "/venv/lib/python3.8/site-packages/chanjo/cli/base.py", line 39, in get_command
    return commands[name].load()
  File "/venv/lib/python3.8/site-packages/pkg_resources/__init__.py", line 2450, in load
    return self.resolve()
  File "/venv/lib/python3.8/site-packages/pkg_resources/__init__.py", line 2456, in resolve
    module = __import__(self.module_name, fromlist=['__name__'], level=0)
  File "/home/worker/app/chanjo_report/cli/__init__.py", line 2, in <module>
    from .core import report
  File "/home/worker/app/chanjo_report/cli/core.py", line 4, in <module>
    from chanjo_report.interfaces import html, pdf
  File "/home/worker/app/chanjo_report/interfaces/__init__.py", line 2, in <module>
    from .html import render_html
  File "/home/worker/app/chanjo_report/interfaces/html.py", line 4, in <module>
    from chanjo_report.server.app import create_app
  File "/home/worker/app/chanjo_report/server/app.py", line 5, in <module>
    from .config import DefaultConfig
  File "/home/worker/app/chanjo_report/server/config.py", line 2, in <module>
    from .blueprints import report_bp, index_bp
  File "/home/worker/app/chanjo_report/server/blueprints/__init__.py", line 3, in <module>
    from .report import report_bp
  File "/home/worker/app/chanjo_report/server/blueprints/report/__init__.py", line 2, in <module>
    from .views import report_bp
  File "/home/worker/app/chanjo_report/server/blueprints/report/views.py", line 7, in <module>
    from flask_weasyprint import HTML, render_pdf
  File "/venv/lib/python3.8/site-packages/flask_weasyprint/__init__.py", line 5, in <module>
    import weasyprint
  File "/venv/lib/python3.8/site-packages/weasyprint/__init__.py", line 341, in <module>
    from .css import preprocess_stylesheet  # noqa isort:skip
  File "/venv/lib/python3.8/site-packages/weasyprint/css/__init__.py", line 25, in <module>
    from . import computed_values, counters, media_queries
  File "/venv/lib/python3.8/site-packages/weasyprint/css/computed_values.py", line 11, in <module>
    from ..text.ffi import ffi, pango, units_to_double
  File "/venv/lib/python3.8/site-packages/weasyprint/text/ffi.py", line 8, in <module>
    ffi = cffi.FFI()
  File "/venv/lib/python3.8/site-packages/cffi/api.py", line 48, in __init__
    import _cffi_backend as backend
ModuleNotFoundError: No module named '_cffi_backend'
make: *** [report] Error 1
```

Any simple fix? Otherwise I think I'll stick with the local db. 😊

**Comment by northwestwitch** [comment]
Weird, it works for me:

image


What do we do different?

**Comment by dnil** [comment]
> Weird, it works for me:
> 
> image
> 
> What do we do different?

Hm odd; are you still on Intel, or could this be some Docker-on-Apple issue? From the error I would expect it needs installing cffi, but that is on the requirements, so one would hope the build would fail if it didn't work. I'll see how it works with local.

**Comment by northwestwitch** [comment]
> Hm odd; are you still on Intel, or could this be some Docker-on-Apple issue? From the error I would expect it needs installing cffi, but that is on the requirements, so one would hope the build would fail if it didn't work. I'll see how it works with local.

Same computer as always. Yes, the lib should be installed with the requirements.

What if you:
from docker desktop:
- remove all related containers
- remove all related images

Then from command line:
- docker system prune
- docker volume prune

?

That's what I did just now and it still works. tested both master and this branch

**Comment by dnil** [comment]
Sure, will try. But docker desktop does flag these as AMD64-builds. Dunno what we have in there that is binary, but maybe some of the rendering stuff.

**Comment by northwestwitch** [comment]
From the error message it looks like a weasyprint related error (it is still haunting us!)

Why should it work differently on our computers if it's docker based, anyway? That's literally the reason why you use docker 🤔

**Comment by northwestwitch** [comment]
Could be some settings in the docker desktop. I don't have those warnings

**Comment by northwestwitch** [comment]
https://forums.docker.com/t/amd64-error-for-silicon-m1/118085/5

**Comment by dnil** [comment]
Agreed, that is silly. I suppose it exposes some of the black magic used to make it work with relative little effort. To me it seems they include some components in some kind of vm-code, some binary-emulation format, and translate other parts where possible with system calls or calls for local libs.

Build was ok, but by the way, setup actually failed on try 1 with a missing table. Seems to work on the second run though.

```
 => [stage-1 8/8] RUN pip install --editable .                                                                                                                                                      1.2s
 => exporting to image                                                                                                                                                                              0.5s
 => => exporting layers                                                                                                                                                                             0.5s
 => => writing image sha256:d7055376acfc4b58ff56ff3b0016fa00163ae4f25ff4aae01760ced3bf5d9f63                                                                                                        0.0s
 => => naming to docker.io/library/chanjo-report-chanjo-report                                                                                                                                      0.0s
➜  chanjo-report git:(master) ✗ make setup
echo "Setup chanjo database"
Setup chanjo database
docker-compose run chanjo-cli /bin/bash -c "chanjo -d mysql+pymysql://chanjoUser:chanjoPassword@mariadb/chanjo4_test init --auto demodata && chanjo --config demodata/chanjo.yaml link demodata/hgnc.grch37p13.exons.bed"
[+] Running 9/9
 ⠿ mariadb Pulled                                                                                                                                                                                   6.9s
   ⠿ 986c4f6be307 Pull complete                                                                                                                                                                     2.4s
   ⠿ 1899f9c38296 Pull complete                                                                                                                                                                     2.4s
   ⠿ 30b0d8350f0a Pull complete                                                                                                                                                                     2.7s
   ⠿ 940b0462cbe1 Pull complete                                                                                                                                                                     2.7s
   ⠿ 74b0786f7c6f Pull complete                                                                                                                                                                     2.8s
   ⠿ d1ff0ed01cba Pull complete                                                                                                                                                                     4.9s
   ⠿ c4b4f00e4728 Pull complete                                                                                                                                                                     5.0s
   ⠿ d38eb59de3cf Pull complete                                                                                                                                                                     5.0s
[+] Running 2/2
 ⠿ Network chanjo-report_chanjo-net  Created                                                                                                                                                        0.0s
 ⠿ Container mariadb                 Created                                                                                                                                                        0.4s
[+] Running 1/1
 ⠿ Container mariadb  Started                                                                                                                                                                       0.2s
[+] Running 13/13
 ⠿ chanjo-cli Pulled                                                                                                                                                                                9.1s
   ⠿ 540db60ca938 Pull complete                                                                                                                                                                     0.7s
   ⠿ 2cf6782fdab7 Pull complete                                                                                                                                                                     0.9s
   ⠿ 68b9956cbc5d Pull complete                                                                                                                                                                     4.0s
   ⠿ dd11062c48b6 Pull complete                                                                                                                                                                     6.0s
   ⠿ 359444161c21 Pull complete                                                                                                                                                                     6.7s
   ⠿ 8dbd338612c1 Pull complete                                                                                                                                                                     7.2s
   ⠿ 53d79be7f4fe Pull complete                                                                                                                                                                     7.3s
   ⠿ ba0138777508 Pull complete                                                                                                                                                                     7.3s
   ⠿ b5a7f4670003 Pull complete                                                                                                                                                                     7.5s
   ⠿ 663c340b80a0 Pull complete                                                                                                                                                                     7.6s
   ⠿ e8f320886b19 Pull complete                                                                                                                                                                     7.6s
   ⠿ 9f8c2c5eaac5 Pull complete                                                                                                                                                                     7.6s
2023-03-06 10:28:43 b0e9e2fd4de9 chanjo.cli.init[8] INFO setting up chanjo under: demodata
2023-03-06 10:28:43 b0e9e2fd4de9 chanjo.init.bootstrap[8] INFO downloading... [https://s3.eu-central-1.amazonaws.com/clinical-assets/hgnc.grch37p13.exons.bed.zip]
2023-03-06 10:28:43 b0e9e2fd4de9 chanjo.init.bootstrap[8] INFO extracting BED file...
2023-03-06 10:28:44 b0e9e2fd4de9 chanjo.init.bootstrap[8] INFO removing BED archive...
2023-03-06 10:28:44 b0e9e2fd4de9 chanjo.cli.init[8] INFO configure new chanjo database: mysql+pymysql://chanjoUser:chanjoPassword@mariadb/chanjo4_test
2023-03-06 10:28:44 b0e9e2fd4de9 chanjo.store.api[8] INFO created tables: transcript, sample, transcript_stat
2023-03-06 10:28:44 b0e9e2fd4de9 chanjo.cli.init[8] INFO writing config file: demodata/chanjo.yaml
Chanjo bootstrap successful! Now run:
chanjo --config demodata/chanjo.yaml link demodata/hgnc.grch37p13.exons.bed
adding transcripts  [####################################]  100%
echo "Loading coverage from demo files"
Loading coverage from demo files
docker-compose run chanjo-cli chanjo -d mysql+pymysql://chanjoUser:chanjoPassword@mariadb/chanjo4_test load -n sample1 --group-name test_group -g test_group chanjo/init/demo-files/sample1.coverage.bed
[+] Running 1/1
 ⠿ Container mariadb  Recreated                                                                                                                                                                     0.5s
[+] Running 1/1
 ⠿ Container mariadb  Started                                                                                                                                                                       0.2s
loading transcripts  [####################################]  100%
Traceback (most recent call last):
  File "/opt/conda/lib/python3.7/site-packages/sqlalchemy/engine/base.py", line 1277, in _execute_context
    cursor, statement, parameters, context
  File "/opt/conda/lib/python3.7/site-packages/sqlalchemy/engine/default.py", line 608, in do_execute
    cursor.execute(statement, parameters)
  File "/opt/conda/lib/python3.7/site-packages/pymysql/cursors.py", line 148, in execute
    result = self._query(query)
  File "/opt/conda/lib/python3.7/site-packages/pymysql/cursors.py", line 310, in _query
    conn.query(q)
  File "/opt/conda/lib/python3.7/site-packages/pymysql/connections.py", line 548, in query
    self._affected_rows = self._read_query_result(unbuffered=unbuffered)
  File "/opt/conda/lib/python3.7/site-packages/pymysql/connections.py", line 775, in _read_query_result
    result.read()
  File "/opt/conda/lib/python3.7/site-packages/pymysql/connections.py", line 1156, in read
    first_packet = self.connection._read_packet()
  File "/opt/conda/lib/python3.7/site-packages/pymysql/connections.py", line 725, in _read_packet
    packet.raise_for_error()
  File "/opt/conda/lib/python3.7/site-packages/pymysql/protocol.py", line 221, in raise_for_error
    err.raise_mysql_exception(self._data)
  File "/opt/conda/lib/python3.7/site-packages/pymysql/err.py", line 143, in raise_mysql_exception
    raise errorclass(errno, errval)
pymysql.err.ProgrammingError: (1146, "Table 'chanjo4_test.sample' doesn't exist")

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/opt/conda/bin/chanjo", line 33, in <module>
    sys.exit(load_entry_point('chanjo', 'console_scripts', 'chanjo')())
  File "/opt/conda/lib/python3.7/site-packages/click/core.py", line 1137, in __call__
    return self.main(*args, **kwargs)
  File "/opt/conda/lib/python3.7/site-packages/click/core.py", line 1062, in main
    rv = self.invoke(ctx)
  File "/opt/conda/lib/python3.7/site-packages/click/core.py", line 1668, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/opt/conda/lib/python3.7/site-packages/click/core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/opt/conda/lib/python3.7/site-packages/click/core.py", line 763, in invoke
    return __callback(*args, **kwargs)
  File "/opt/conda/lib/python3.7/site-packages/click/decorators.py", line 26, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/home/worker/app/chanjo/cli/load.py", line 55, in load
    chanjo_db.save()
  File "/home/worker/app/chanjo/store/api.py", line 124, in save
    raise error
  File "/home/worker/app/chanjo/store/api.py", line 120, in save
    self.commit()
  File "/opt/conda/lib/python3.7/site-packages/sqlalchemy/orm/scoping.py", line 163, in do
    return getattr(self.registry(), name)(*args, **kwargs)
  File "/opt/conda/lib/python3.7/site-packages/sqlalchemy/orm/session.py", line 1046, in commit
    self.transaction.commit()
  File "/opt/conda/lib/python3.7/site-packages/sqlalchemy/orm/session.py", line 504, in commit
    self._prepare_impl()
  File "/opt/conda/lib/python3.7/site-packages/sqlalchemy/orm/session.py", line 483, in _prepare_impl
    self.session.flush()
  File "/opt/conda/lib/python3.7/site-packages/sqlalchemy/orm/session.py", line 2540, in flush
    self._flush(objects)
  File "/opt/conda/lib/python3.7/site-packages/sqlalchemy/orm/session.py", line 2682, in _flush
    transaction.rollback(_capture_exception=True)
  File "/opt/conda/lib/python3.7/site-packages/sqlalchemy/util/langhelpers.py", line 70, in __exit__
    with_traceback=exc_tb,
  File "/opt/conda/lib/python3.7/site-packages/sqlalchemy/util/compat.py", line 182, in raise_
    raise exception
  File "/opt/conda/lib/python3.7/site-packages/sqlalchemy/orm/session.py", line 2642, in _flush
    flush_context.execute()
  File "/opt/conda/lib/python3.7/site-packages/sqlalchemy/orm/unitofwork.py", line 422, in execute
    rec.execute(self)
  File "/opt/conda/lib/python3.7/site-packages/sqlalchemy/orm/unitofwork.py", line 589, in execute
    uow,
  File "/opt/conda/lib/python3.7/site-packages/sqlalchemy/orm/persistence.py", line 245, in save_obj
    insert,
  File "/opt/conda/lib/python3.7/site-packages/sqlalchemy/orm/persistence.py", line 1083, in _emit_insert_statements
    c = cached_connections[connection].execute(statement, multiparams)
  File "/opt/conda/lib/python3.7/site-packages/sqlalchemy/engine/base.py", line 1011, in execute
    return meth(self, multiparams, params)
  File "/opt/conda/lib/python3.7/site-packages/sqlalchemy/sql/elements.py", line 298, in _execute_on_connection
    return connection._execute_clauseelement(self, multiparams, params)
  File "/opt/conda/lib/python3.7/site-packages/sqlalchemy/engine/base.py", line 1130, in _execute_clauseelement
    distilled_params,
  File "/opt/conda/lib/python3.7/site-packages/sqlalchemy/engine/base.py", line 1317, in _execute_context
    e, statement, parameters, cursor, context
  File "/opt/conda/lib/python3.7/site-packages/sqlalchemy/engine/base.py", line 1511, in _handle_dbapi_exception
    sqlalchemy_exception, with_traceback=exc_info[2], from_=e
  File "/opt/conda/lib/python3.7/site-packages/sqlalchemy/util/compat.py", line 182, in raise_
    raise exception
  File "/opt/conda/lib/python3.7/site-packages/sqlalchemy/engine/base.py", line 1277, in _execute_context
    cursor, statement, parameters, context
  File "/opt/conda/lib/python3.7/site-packages/sqlalchemy/engine/default.py", line 608, in do_execute
    cursor.execute(statement, parameters)
  File "/opt/conda/lib/python3.7/site-packages/pymysql/cursors.py", line 148, in execute
    result = self._query(query)
  File "/opt/conda/lib/python3.7/site-packages/pymysql/cursors.py", line 310, in _query
    conn.query(q)
  File "/opt/conda/lib/python3.7/site-packages/pymysql/connections.py", line 548, in query
    self._affected_rows = self._read_query_result(unbuffered=unbuffered)
  File "/opt/conda/lib/python3.7/site-packages/pymysql/connections.py", line 775, in _read_query_result
    result.read()
  File "/opt/conda/lib/python3.7/site-packages/pymysql/connections.py", line 1156, in read
    first_packet = self.connection._read_packet()
  File "/opt/conda/lib/python3.7/site-packages/pymysql/connections.py", line 725, in _read_packet
    packet.raise_for_error()
  File "/opt/conda/lib/python3.7/site-packages/pymysql/protocol.py", line 221, in raise_for_error
    err.raise_mysql_exception(self._data)
  File "/opt/conda/lib/python3.7/site-packages/pymysql/err.py", line 143, in raise_mysql_exception
    raise errorclass(errno, errval)
sqlalchemy.exc.ProgrammingError: (pymysql.err.ProgrammingError) (1146, "Table 'chanjo4_test.sample' doesn't exist")
[SQL: INSERT INTO sample (id, group_id, source, created_at, name, group_name) VALUES (%(id)s, %(group_id)s, %(source)s, %(created_at)s, %(name)s, %(group_name)s)]
[parameters: {'id': 'sample1', 'group_id': 'test_group', 'source': '/home/worker/app/chanjo/init/demo-files/sample1.coverage.bed', 'created_at': datetime.datetime(2023, 3, 6, 10, 30, 1, 356548), 'name': 'sample1', 'group_name': 'test_group'}]
(Background on this error at: http://sqlalche.me/e/13/f405)
make: *** [setup] Error 1
```

**Comment by northwestwitch** [comment]
Check out this one as well: https://stackoverflow.com/questions/65612411/forcing-docker-to-use-linux-amd64-platform-by-default-on-macos

**Comment by dnil** [comment]
Ah, both the first pass setup issue and the overall problem resolved when forcing AMD64 with 
```
export DOCKER_DEFAULT_PLATFORM=linux/amd64
```
as per the second link! Looking promising!
Screenshot 2023-03-06 at 11 21 26

**Comment by northwestwitch** [comment]
Perhaps we can introduce the env var in the docker-compose fileS, to it's fixed for anyone having a M1 arm processor?

**Comment by dnil** [comment]
Agreed!

**Comment by northwestwitch** [comment]
The PR looks fine now I think, but it contained code changes that belonged to other old PRs. I've fixed by merging master into it!

**Review by northwestwitch — APPROVED** [review]
Looks good and runs fine, so I think it is fine to merge.

I'm not 100% sure that this PR will fix [our issue](https://github.com/Clinical-Genomics/scout/issues/2897) but let's try!

I'm saying this because I remember seeing another type of error message when session object was too big (session cookie too large or something). Why did you suspect that this was the reason of our error in the first place @dnil?

**Comment by dnil** [comment]
If I recall correctly the first clue was the dialog with IE that she was kicked out immediately after requesting a coverage report on a case with a big gene list. Then we have seen it repeatedly with several users, and clearing cookies always solves it. This is one of the few places I know of where we explicitly store on the Scout Session. I think we will still be vulnerable to a similar issue as long as we rely on single-cookie storage, but as long as we don't get like overly long file paths for IGV...

**Comment by dnil** [comment]
But you are right, it is reasoning by correlation, though with a mechanism for why. Perhaps we can try to reproduce on scout-stage; that would make feel closing the issue more ok?

**Comment by northwestwitch** [comment]
> But you are right, it is reasoning by correlation, though with a mechanism for why. Perhaps we can try to reproduce on scout-stage; that would make feel closing the issue more ok?

I think it is fine to merge and see if the problem is solved. If it isn't we can find another solution!

**Comment by dnil** [comment]
Ok, let's see! It is a little hard to reproduce the whole thing on local. A docker-compose might do it, but... I'll merge then, and we can test it on stage.

**Comment by dnil** [comment]
Ok, deployed to stage by rebuilding a branch, and making sure 4.10.1 is in.
Screenshot 2023-03-10 at 14 25 00
Screenshot 2023-03-10 at 14 25 00

So far, so good from just clicking around in chanjo-report, with big and small gene lists.

**Pull request merged**